broker/lib/controllers/ServiceBrokerApiController.js

```diff
@@ -105,22 +173,89 @@ class ServiceBrokerApiController extends FabrikBaseController {
       .then(done);
   }
 
+  patchInstanceV2(req, res) {
+    const params = _
+      .chain(req.body)
+      .omit('plan_id', 'service_id')
+      .cloneDeep()
+      .value();
+    //cloning here so that the DirectorInstance.update does not unset the 'service-fabrik-operation' from original req.body object
+
+    function done(result) {
+      let statusCode = CONST.HTTP_STATUS_CODE.OK;
+      const body = {};
+      if (req.instance.async) {
+        statusCode = CONST.HTTP_STATUS_CODE.ACCEPTED;
+        body.operation = utils.encodeBase64(result);
+      } else if (result && result.description) {
+        body.description = result.description;
+      }
+      res.status(statusCode).send(body);
+    }
+
+    req.operation_type = CONST.OPERATION_TYPE.UPDATE;
+    this.validateRequest(req, res);
+    let lockedDeployment = false;
+    return Promise
+      .try(() => {
+        if (!req.manager.isUpdatePossible(params.previous_values.plan_id)) {
+          throw new BadRequest(`Update to plan '${req.manager.plan.name}' is not possible`);
+        }
+        return Promise.try(() => {
+            if (req.manager.name === CONST.INSTANCE_TYPE.DIRECTOR) {
+              // Acquire lock for this instance
+              return lockManager.lock(req.params.instance_id, {
+                lockType: CONST.ETCD.LOCK_TYPE.WRITE,
+                lockedResourceDetails: {
+                  resourceType: CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT,
+                  resourceName: CONST.APISERVER.RESOURCE_NAMES.DIRECTOR,
+                  resourceId: req.params.instance_id,
+                  operation: CONST.OPERATION_TYPE.UPDATE
+                }
+              });
+            }
+          })
+          .then(() => {
+            lockedDeployment = true;
+            return req.instance.update(params);
+          });
+      })
+      .then(done)
+      .catch(err => {
+        if (err instanceof EtcdLockError) {
+          throw err;
+        }
+        if (lockedDeployment) {
```

Comment:
Why this flag is needed? Can't `lockManager.unlock` handle the case where lock was not obtained? Anyways the goal is to unlock, if it was not locked in the first place, doesn't matter here.
If this flag is needed, then why same is not added in [putInstanceV2](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/304/files#diff-03e33ae33c13c68e445aa8cc1048acc9R130) and [deleteInstanceV2](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/304/files#diff-03e33ae33c13c68e445aa8cc1048acc9R308)?

Reply:
`lockManager.unlock` handles unlocking if lock is not present. Here we need to know if error occurred before acquiring the lock (ex: [here](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/304/files#diff-03e33ae33c13c68e445aa8cc1048acc9R201)), if it did then we should not unlock resource.
Not present in `putInstance` and some other places as there acquiring the lock is the first call that we are making.
Putting comments in the code explaining the same